File: java/duck/ducks/MallardDuck.java
package ducks;

import behaviors.fly.FlyWithWings;
import behaviors.quack.Quack;

public class MallardDuck extends Duck {

    public MallardDuck(String name) {
        super(name);
        quackBehavior = new Quack(name);
        flyBehavior = new FlyWithWings(name);
    }
}

File: kotlin/duck/ducks/MallardDuck.kt
package ducks

import behaviors.fly.FlyWithWings
import behaviors.quack.Quack

class MollardDuck(name: String) : Duck(name) {

    init {
        quackBehavior = Quack(name)
        flyBehavior = FlyWithWings(name)
    }
}